src/core/include/openvino/pass/manager.hpp

```diff
@@ -66,11 +69,8 @@ class OPENVINO_API Manager {
     ///
     /// \return     Returns true if the model was changed by transformations,
     ///             false otherwise.
-    bool run_passes(std::shared_ptr<Model> model);
+    bool run_passes(const std::shared_ptr<Model>& model);
```

Comment:
```suggestion
    bool run_passes(const Model& model);
```
And use just model inside?
Or anything is done with poitner ownership, previous version indicated that run_passe will take owner ship also.

Reply:
>>  bool run_passes(const Model& model);
in this case, we will have to update other classes, e.g. ModelPass, or use shared_from_this
I would prefer not to change it in this PR.
I believe this is a common mistake of openvino project, we prefer to use const shared_ptr<T>& instead of const T& in most places.

About ownership: it looks like ModelPass and GraphRewrite, which are called inside Manager, already accept `const shared_ptr<T>&` and we don't need to copy  shared_ptr<> and increment the ref count in `run_passes`. `run_passes` is called from single thread, and outer scope waits when it finishes the execution, the  real shared ptr is stored outside.